metric/cpu/metrics_windows.go

```diff
@@ -27,12 +27,68 @@ import (
 	"time"
 
 	"github.com/elastic/elastic-agent-libs/opt"
-	"github.com/elastic/elastic-agent-system-metrics/metric/system/resolve"
 	"github.com/elastic/gosigar/sys/windows"
 )
 
+var (
+	processorInformationCounter = "\\Processor Information(%s)\\%s"
+	totalKernelTimeCounter      = fmt.Sprintf(processorInformationCounter, "*", "% Privileged Time")
+	totalIdleTimeCounter        = fmt.Sprintf(processorInformationCounter, "*", "% Idle Time")
+	totalUserTimeCounter        = fmt.Sprintf(processorInformationCounter, "*", "% User Time")
+)
+
 // Get fetches Windows CPU system times
-func Get(_ resolve.Resolver) (CPUMetrics, error) {
+func Get(m *Monitor) (CPUMetrics, error) {
+	if m.query == nil {
+		return defaultGet()
+	}
```

Comment:
Nit.  Just to improve readability and testing.  I think it would be nice if `Get` was something like:

```
func Get(m *Monitor) (CPUMetrics, error) {
    if m.query == nil {
        return getUsingSystemTimes()
    }
    return getUsingPerformanceCounters(m)
}
```

That way it is explicit that the only thing Get does is pick which implementation to use, and it is easy to test the 2 implementations independently.

Reply:
Got it. Thanks!